### PR TITLE
Geonode connection dialog

### PR DESCRIPTION
**Pull request by myarjunar**
## Description
Add general functionality to the manager dialog. We still cannot add layer to the map.

Create new geonode connection:
create-new-geonode-connection

Save geonode connection:
save-geonode-connection

Load geonode connection from exported connection:
load-geonode-connection

Delete geonode connection:
delete-geonode-connection

### CODE REVIEW

File: src/app/geonode/qgsgeonodesourceselect.cpp [hunk 1]
```diff
@@ -40,10 +49,219 @@ QgsGeonodeSourceSelect::QgsGeonodeSourceSelect( QWidget *parent, Qt::WindowFlags
...
+  for ( int i = MODEL_IDX_TITLE; i < MODEL_IDX_TYPE; i++ )
+  {
+    if ( treeView->columnWidth( i ) > 300 )
+    {
+      treeView->setColumnWidth( i, 300 );
+    }
+  }
+}
+
+void QgsGeoNodeSourceSelect::on_btnSave_clicked()
```
Comment (Gustry) [inline src/app/geonode/qgsgeonodesourceselect.cpp]: AFAIK, you need to use the `connect` function and not these auto-connect slots.

File: src/app/geonode/qgsgeonodesourceselect.cpp [hunk 2]
```diff
@@ -40,10 +49,219 @@ QgsGeonodeSourceSelect::QgsGeonodeSourceSelect( QWidget *parent, Qt::WindowFlags
...
+{
+  QgsManageConnectionsDialog dlg( this, QgsManageConnectionsDialog::Export, QgsManageConnectionsDialog::GeoNode );
+  dlg.exec();
+}
+
+void QgsGeoNodeSourceSelect::on_btnLoad_clicked()
```
Comment (Gustry) [inline src/app/geonode/qgsgeonodesourceselect.cpp]: Same here

File: src/app/geonode/qgsgeonodesourceselect.cpp (hunk 1 above)
Comment (myarjunar) [inline src/app/geonode/qgsgeonodesourceselect.cpp]: get rid of legacy code?

Comment (Gustry) [inline src/app/geonode/qgsgeonodesourceselect.cpp]: Yes, this is legacy. QGIS Developers are removing them step by step.
Having an explicit connection is better.

**Comment by Gustry** [comment]
* I'm just thinking that we should have a checkbox: `open selected layers in group` if you have more than one layers selected and use the name of the geonode instance to create a group.

* What will do the 'build query' button? I know for WFS you can give a filter option. But I think let's keep it simple for now and not use it.

**Comment by ismailsunni** [comment]
Thanks @myarjunar 
I thought we want to make a PR from geonode-client branch to kartoza/QGIS master branch (and update it regularly from the QGIS/QGIS master).

**Comment by Gustry** [comment]
Yes, we are missing some code in this PR

**Pull request merged**